File: Cpp/1128_Number_of_Equivalent_Domino_Pairs.cpp
class Solution {
public:
    int numEquivDominoPairs(vector<vector<int>>& dominoes) {
        int result = 0;
        
        vector<vector<int>> visit(10, vector<int>(10));
        for(vector<int>& domino : dominoes) {
            int min_value = min(domino[0], domino[1]);
            int max_value = max(domino[0], domino[1]);
            result += visit[min_value][max_value]++;
        }
        
        return result;
    }
};

File: Kotlin/1128_Number_of_Equivalent_Domino_Pairs.kt
class Solution {
    fun numEquivDominoPairs(dominoes: Array<IntArray>): Int {
        var result = 0
        val visit = Array(10) { IntArray(10) }
        
        for(domino in dominoes) {
            val minValue = minOf(domino[0], domino[1])
            val maxValue = maxOf(domino[0], domino[1])
            
            result += visit[minValue][maxValue]++
        }
        
        return result
    }
}